src/automediaembed.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module media-embed/automediaembed
+ */
+
+import MediaEmbedEditing from './mediaembedediting';
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import Clipboard from '@ckeditor/ckeditor5-clipboard/src/clipboard';
+import Range from '@ckeditor/ckeditor5-engine/src/model/range';
+import Position from '@ckeditor/ckeditor5-engine/src/model/position';
+import LivePosition from '@ckeditor/ckeditor5-engine/src/model/liveposition';
+import TreeWalker from '@ckeditor/ckeditor5-engine/src/model/treewalker';
+import global from '@ckeditor/ckeditor5-utils/src/dom/global';
+
+const URL_REGEXP = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w.-]+)+[\w\-._~:/?#[\]@!$&'()*+,;=]+$/;
+
+/**
+ * The auto-media embed plugin.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class AutoMediaEmbed extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ Clipboard ];
```

Comment:
I'm not sure we need that. Importing that is the job of the `Essentials` plugin, which is the key part of each editor.

Reply:
OTOH, it doesn't hurt so I guess it can stay :P

Reply:
You're right when you said:

> I'm not sure we need that. Importing that is the job of the Essentials plugin, which is the key part of each editor.

but we cannot assume that every editor will use the `Essentials` plugin. I'm defining it as a dependency because it's required for `AutoMediaEmbed` to work from its point of view.